packages/snaps-controllers/src/snaps/location/location.ts

```diff
@@ -0,0 +1,64 @@
+import { SnapManifest, VFile } from '@metamask/snaps-utils';
+import { assert } from '@metamask/utils';
+
+import HttpLocation from './http';
+import LocalLocation from './local';
+import NpmLocation, { NpmOptions } from './npm';
+
+declare module '@metamask/snaps-utils' {
```

Comment:
This allows `vfile` to be decoupled from snaps so that it can be put into `@metamask/utils`. Any module that operates on vfiles can declare it's own data properties and they'll be properly typed whenever using vfile inside that module.

For example, `snaps-controllers` can add canonical paths, while `snaps-cli` can add `fixableErrors` metadata instead of throwing.